portality/view/oaipmh.py

```diff
@@ -4,32 +4,44 @@
 from flask import Blueprint, request, make_response
 from portality.core import app
 from portality.models import OAIPMHJournal, OAIPMHArticle
+from portality.lib import analytics
 from portality import datasets
 from copy import deepcopy
 
 blueprint = Blueprint('oaipmh', __name__)
 
 #####################################################################
-## Web API endpoints
+# Web API endpoints
 #####################################################################
 
+
 @blueprint.route("/oai", methods=["GET", "POST"])
 @blueprint.route("/oai.<specified>", methods=["GET", "POST"])
 def oaipmh(specified=None):
+    # Google Analytics event
+    event = {'category': 'OAI-PMH', 'action': None, 'label': None, 'fieldsobject': None}
```

Comment:
I think there's too much knowledge about the internal shape of an Event here.  It would be better if the analytics code presented an Event class which was populated through an interface, then implementers don't need to remember what all the fields are called, or risk making a spelling error, or something like that.